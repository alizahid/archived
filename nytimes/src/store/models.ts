import { Action, Thunk, action, thunk } from 'easy-peasy'

import { api } from '../lib'

export interface Article {
  abstract: string
  byline: string
  id: number
  image: string
  published: any
  thumb: string
  title: string
  url: string
  views: number
}

export interface ArticlesModel {
  interval: number
  loading: boolean

  articles: Article[]

  fetchArticles: Thunk<ArticlesModel, number>

  setInterval: Action<ArticlesModel, number>
  setLoading: Action<ArticlesModel, boolean>
  setArticles: Action<ArticlesModel, Article[]>
}

const articles: ArticlesModel = {
  interval: 1,
  loading: false,

  articles: [],

  fetchArticles: thunk(async (actions, payload) => {
    actions.setLoading(true)

    const articles = await api.fetchArticles(payload)

    actions.setArticles(articles)

    actions.setLoading(false)
  }),

  setInterval: action((state, payload) => {
    state.interval = payload
  }),
  setLoading: action((state, payload) => {
    state.loading = payload
  }),
  setArticles: action((state, payload) => {
    state.articles = payload
  })
}

export interface StoreModel {
  articles: ArticlesModel
}

const model: StoreModel = {
  articles
}

export default model
