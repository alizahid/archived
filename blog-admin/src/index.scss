// fonts

@import url('https://rsms.me/inter/inter.css');
@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Mono:400,700&display=swap');

// reset

* {
  border-radius: 0;
  box-sizing: border-box;
  font-weight: normal;
  margin: 0;
  outline: none;
  padding: 0;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
}

// global

a {
  color: #ff2d55;
  cursor: pointer;
  text-decoration: none;
  transition: 0.2s;

  &:hover {
    filter: brightness(1.2);
  }
}

body {
  background: #fff;
  color: #000;
  margin: 0;
  font: normal 1em/1 'Inter var', -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
    'Helvetica Neue', sans-serif;
  overflow: hidden;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font: normal 1em/1 'IBM Plex Mono', source-code-pro, Menlo, Monaco, Consolas,
    'Courier New', monospace;
  line-height: 1.6;
}

form {
  label {
    display: block;
    margin: 1em 0;

    span {
      display: block;
      font-weight: 500;
      margin-bottom: 1em;
    }
  }

  button,
  input,
  textarea {
    appearance: none;
    background: #fff;
    border: 1px solid #ecf0f1;
    border-radius: 0.25em;
    font: inherit;
    padding: 1em;
    transition: 0.3s;
  }

  button {
    background: #ecf0f1;
    border: none;
    cursor: pointer;
    line-height: normal;
    padding: 1em 2em;

    &:hover {
      background: #bdc3c7;
    }
  }

  input,
  textarea {
    width: 100%;

    &:focus {
      border-color: #bdc3c7;
    }
  }

  textarea {
    line-height: 1.4;
    min-height: 10em;
  }
}

h1,
h2,
h3 {
  font-weight: 600;
  margin: 1em 0;
}

h1 {
  font-size: 4em;
}

h2 {
  font-size: 3em;
  margin-bottom: 0.5em;
}

h3 {
  font-size: 2em;
  margin-bottom: 0.5em;
}

hr {
  border: none;
  height: 1px;
  margin: 2em 0;
  position: relative;
}

hr:before {
  background: linear-gradient(
    to left,
    rgba(0, 0, 0, 0.01),
    rgba(0, 0, 0, 0.05),
    rgba(0, 0, 0, 0.01)
  );
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

img {
  max-width: 100%;
  vertical-align: top;
}

ol,
ul {
  margin: 1em 0;
}

ol li,
ul li {
  line-height: 1.6;
  list-style: square;
  margin: 1em 0 1em 2em;
}

p {
  margin: 1em 0;
  line-height: 1.6;
}

// layout

#root {
  display: flex;
  flex-direction: column;
  height: 100vh;
}
