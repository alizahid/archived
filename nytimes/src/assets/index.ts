export { default as icon_date } from './icons8-calendar.png'
export { default as icon_views } from './icons8-visible.png'
