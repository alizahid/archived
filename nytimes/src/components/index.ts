export { default as Full } from './full'
export { default as Loading } from './loading'
export { default as NotFound } from './not-found'
export { default as Periods } from './periods'
export { default as Preview } from './preview'
export { default as Separator } from './separator'
