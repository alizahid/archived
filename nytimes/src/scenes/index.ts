export { default as Article } from './article'
export { default as Home } from './home'
