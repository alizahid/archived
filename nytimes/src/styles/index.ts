import { iOSColors } from 'react-native-typography'

export const layout = {
  icon: 20,
  image: 50,
  margin: 20,
  padding: 10,
  radius: 5
}

export const shadow = {
  elevation: 5,
  shadowColor: iOSColors.gray,
  shadowOffset: {
    height: 0,
    width: 0
  },
  shadowOpacity: 0.25,
  shadowRadius: 5
}
