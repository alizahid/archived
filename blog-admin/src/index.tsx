import './index.scss'

import React, { FunctionComponent } from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter, Route } from 'react-router-dom'

import { Header } from './components'
import { Auth, Posts, Projects } from './scenes'
import * as serviceWorker from './serviceWorker'

const Blog: FunctionComponent = () => (
  <BrowserRouter>
    <Header />
    <Route path="/" exact component={Auth} />
    <Route path="/posts" component={Posts} />
    <Route path="/projects" component={Projects} />
  </BrowserRouter>
)

ReactDOM.render(<Blog />, document.getElementById('root'))

serviceWorker.unregister()
