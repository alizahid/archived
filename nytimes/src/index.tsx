import React, { FunctionComponent } from 'react'
import { createAppContainer, createStackNavigator } from 'react-navigation'
import { StoreProvider } from 'easy-peasy'

import { Article, Home } from './scenes'
import { store } from './store'

const Navigator = createStackNavigator({
  Home,
  Article
})

const NYTimes: FunctionComponent = () => {
  const Container = createAppContainer(Navigator)

  return (
    <StoreProvider store={store}>
      <Container />
    </StoreProvider>
  )
}

export default NYTimes
