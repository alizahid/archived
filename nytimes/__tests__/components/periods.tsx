import React from 'react'
import { TouchableOpacity } from 'react-native'
import { act, create } from 'react-test-renderer'

import { StoreProvider } from 'easy-peasy'

import Periods from '../../src/components/periods'
import { store } from '../../src/store'

it('renders', () => {
  const tree = create(
    <StoreProvider store={store}>
      <Periods />
    </StoreProvider>
  )

  expect(tree).toMatchSnapshot()
})

it('tap', () => {
  const tree = create(
    <StoreProvider store={store}>
      <Periods />
    </StoreProvider>
  )

  act(() => {
    tree.root.findAllByType(TouchableOpacity)[0].props.onPress(1)
  })

  expect(tree).toMatchSnapshot()
})
