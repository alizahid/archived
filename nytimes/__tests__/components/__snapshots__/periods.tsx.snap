// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "elevation": 5,
      "shadowColor": "#8E8E93",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "padding": 20,
        "paddingLeft": 0,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4CD964",
          "borderRadius": 5,
          "marginLeft": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
              "letterSpacing": -0.408,
              "lineHeight": 22,
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Today
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "marginLeft": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
              "letterSpacing": -0.408,
              "lineHeight": 22,
            },
            false,
          ]
        }
      >
        Last week
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "marginLeft": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
              "letterSpacing": -0.408,
              "lineHeight": 22,
            },
            false,
          ]
        }
      >
        Last month
      </Text>
    </View>
  </View>
</SafeAreaView>
`;

exports[`tap 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "elevation": 5,
      "shadowColor": "#8E8E93",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "padding": 20,
        "paddingLeft": 0,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4CD964",
          "borderRadius": 5,
          "marginLeft": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
              "letterSpacing": -0.408,
              "lineHeight": 22,
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Today
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "marginLeft": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
              "letterSpacing": -0.408,
              "lineHeight": 22,
            },
            false,
          ]
        }
      >
        Last week
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "marginLeft": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
              "letterSpacing": -0.408,
              "lineHeight": 22,
            },
            false,
          ]
        }
      >
        Last month
      </Text>
    </View>
  </View>
</SafeAreaView>
`;
